superset/assets/src/explore/components/controls/AnnotationLayer.jsx

```diff
@@ -331,6 +332,16 @@ export default class AnnotationLayer extends React.PureComponent {
     this.props.close();
   }
 
+  renderOption(option) {
```

Comment:
Is it possible to use styling to elide the text such as "text-overflow"

Reply:
ooo, that's much better. Updated

Reply:
I was wondering the same thing. The JS-based character number may still exhibit issues if the widths of characters are wide enough to cause a linewrap. It might be solvable with just some CSS like:
```
.cleverSpanSelectorHere {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
```